torchao/float8/config.py

```diff
@@ -62,6 +62,7 @@ class CastConfig:
     scaling_type: ScalingType = ScalingType.DYNAMIC
     scaling_granularity: ScalingGranularity = ScalingGranularity.TENSORWISE
     static_scale: Optional[torch.Tensor] = None
+    dtype: Optional[torch.dtype] = None
```

Comment:
nit:
1. can we add a comment on what this is used for, and that `None` means the default e4m3|e5m2 value will be used?
2. optional - thoughts about naming this in a more specific way such as `target_dtype`, `lowp_dtype`, etc?  `dtype` is a bit ambiguous across torchao unfortunately :(